File: Android App/app/src/main/java/revert/shredsheets/models/scales/Neopolitan.java
package revert.shredsheets.models.scales;

public class Neopolitan extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{ 1, 2, 2, 2, 2, 2, 1};
    }

    @Override
    public String getName() {
        return "Neopolitan";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{"Neopolitan", "Leading Whole-Tone", "Lydian Augmented Dominant", "Lydian Dominant ♭6", "Major Locrian", "SemiLocrian ♭4", "SuperLocrian ♭♭3"};
    }

    @Override
    public int[][] getDegrees() {
        return new int[][] {
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 }
        };
    }

    @Override
    public String[] getChords() {
        return new String[] { "mΔ7", "M7+", "M+ ♭7", "M ♭7", "7 ♭5", "m7 ♭5", "♭♭3 ♭5 ♭7" };
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/Neopolitan.kt
package com.bm.shredsheets.models.scales

class Neopolitan : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(1, 2, 2, 2, 2, 2, 1)

    override val name: String
        get() = "Neopolitan"

    override fun getModeNames(): Array<String> {
        return arrayOf("Neopolitan", "Leading Whole-Tone", "Lydian Augmented Dominant", "Lydian Dominant ♭6", "Major Locrian", "SemiLocrian ♭4", "SuperLocrian ♭♭3")
    }

    override val getEnum: Scales
        get() = Scales.Neopolitan

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7))

    override val chords: Array<String>
        get() = arrayOf("mΔ7", "M7+", "M+ ♭7", "M ♭7", "7 ♭5", "m7 ♭5", "♭♭3 ♭5 ♭7")
}